crates/sui-framework/sources/governance/staking_pool.move

```diff
@@ -271,25 +280,57 @@ module sui::staking_pool {
 
     // ==== inactive pool related ====
 
-    /// Deactivate a staking pool by wrapping it in an `InactiveStakingPool` and sharing this newly created object.
-    /// After this pool deactivation, the pool stops earning rewards. Only delegation withdraws can be made to the pool.
-    public(friend) fun deactivate_staking_pool(pool: StakingPool, ctx: &mut TxContext) {
-        let inactive_pool = InactiveStakingPool { id: object::new(ctx), pool};
-        transfer::share_object(inactive_pool);
+    /// Deactivate a staking pool by setting the `deactivation_epoch`. After
+    /// this pool deactivation, the pool stops earning rewards. Only delegation
+    /// withdraws can be made to the pool.
+    public(friend) fun deactivate_staking_pool(pool: StakingPool, ctx: &mut TxContext): StakingPool {
+        // We can't deactivate an already deactivated pool.
+        assert!(pool_active(&pool), EPoolInactive);
```

Comment:
Might be useful to provide separate error codes for trying to delegate to an inactive pool and attempting to deactive an inactive pool, since these likely require different mitigations on the user side.